dask/array/core.py

```diff
@@ -2374,11 +2385,37 @@ def store(x, index, lock):
     slices = slices_from_chunks(arr.chunks)
 
     name = 'store-%s' % arr.name
-    dsk = dict(((name,) + t[1:], (store, t, slc, lock))
+    dsk = dict(((name,) + t[1:], (store, t, slc, lock, region))
                for t, slc in zip(core.flatten(arr._keys()), slices))
     return dsk
 
 
+def storeindask(out, index, x):
+    #Loop through output chunks and setitem:
+    out_sliced = out[index]
+    for slc in slices_from_chunks(out_sliced.chunks):
+        dsk = optimize(out_sliced[slc].dask, out_sliced[slc]._keys())
+        if len(dsk) == 1:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[key][1], dsk[key][2], x[slc])
+        elif len(dsk) == 2:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[dsk[key][1]], dsk[key][2], x[slc])
+        else:
+            raise NotImplementedError('Only concantenated dask arrays can be stored to')
+
+
+def base_condition(dsk, key):
+    #Simple condition that allows simple setitem:
+    return (isinstance(key, tuple) and
+            isinstance(dsk[key],tuple) and
+            len(dsk[key]) == 3 and
+            dsk[key][0] is getarray and
+            isinstance(dsk[key][2], tuple))
```

Comment:
That would definitely work. What I had in mind thought was a workflow of the like:
```
x = da.from_array(...)
y = x.coarsen(np.mean, 10)

files = [h5py.File(fn,'a') for fn in filenames]
dset = da.concatenate([da.from_array(f['x'], chunks=f['x'].chunks)
                       for f in files], axis=0)
da.store(x, dset)

dset1 = h5py.File(final_filename1, 'w')
da,.store(dset.coarsen(np.mean, 5), dset1['x1'])
...
dsetN = h5py.File(final_filenameN, 'w')
da,.store(dset.coarsen(np.var ,5), dset1['xN'])
```
I feels more natural to me but if, as you said, this is not something you want to support, it's fine by me!